kratos/solving_strategies/schemes/scheme.h

```diff
@@ -511,6 +515,16 @@ class Scheme
      * @param rCurrentElement The element to compute
      * @param rCurrentProcessInfo The current process info instance
      */
+    virtual void InitializeNonLinearIteration(
```

Comment:
is this function even necessary on the scheme?
If so then we should probably also add the version for the Constraints

Reply:
this should be still clarified, but can be done in a future PR